### PR TITLE
Bump service bus package

### PR DESCRIPTION
**Pull request by mablae**
WIP

### CODE REVIEW

**Comment by codeliner** [comment]
travis is red. travis.yml needs update like https://github.com/prooph/event-store/blob/develop/.travis.yml#L6

**Comment by codeliner** [comment]
some updates in the package are required, too.
The PluginsPass will no longer work:
https://github.com/prooph/service-bus-symfony-bundle/blob/master/src/DependencyInjection/Compiler/PluginsPass.php#L39

New set up is:
`$plugin->attachToMessageBus($messageBus)`
See: https://github.com/prooph/service-bus/blob/develop/src/Container/AbstractBusFactory.php#L133

So we need something like we already use in the event-store-symfony-bundle:
https://github.com/prooph/event-store-symfony-bundle/blob/master/src/DependencyInjection/Compiler/PluginsPass.php
and
https://github.com/prooph/event-store-symfony-bundle/blob/master/src/EventStoreFactory.php#L31

Unfortunately, a lot of changes required for the service-bus-bundle :( 
@mablae Do you want to continue? Let me know if something is not clear.

**Comment by mablae** [comment]
@codeliner Thanks for your valuable feedback. I will continue with this PR tonight and then look into 
event-store-bundle, too. 

Also it should be a new v2.0.0 instead of a v0.3 version with prooph v7 I guess? 

This PR was not meant to be merged yet, I forgot to add the `WIP`

**Comment by prolic** [comment]
v0.3 is correct

**Comment by UFOMelkor** [comment]
How about tagging the current version as v0.3 and the prooph-v7 one as v0.4?

**Comment by prolic** [comment]
We can do that.

Am 04.01.2017 7:49 nachm. schrieb "Oskar Pfeifer-Bley" <
notifications@github.com>:

**Comment by prolic** [comment]
v0.3 released: https://github.com/prooph/service-bus-symfony-bundle/releases/tag/v0.3

**Comment by UFOMelkor** [comment]
Thx :-)

**Comment by mablae** [comment]
@codeline @prolic I have updated and rebased everything. This PR depends on https://github.com/prooph/service-bus/pull/151

File: .travis.yml [hunk 1]
```diff
@@ -14,29 +14,27 @@ cache:
...
       env:
+        - DEPENDENCIES=""
         - EXECUTE_CS_CHECK=true
-        - EXECUTE_TEST_COVERALLS=true
+        - TEST_COVERAGE=true
 
 before_install:
   - if [[ $EXECUTE_TEST_COVERALLS != 'true' ]]; then phpenv config-rm xdebug.ini || return 0 ; fi
-  - composer self-update
+  - composer update --prefer-dist $DEPENDENCIES
```
Comment (UFOMelkor) [inline .travis.yml]: Replacement of `self-update` with `update` intended?

File: .travis.yml [hunk 2]
```diff
@@ -14,29 +14,27 @@ cache:
...
   - if [[ $EXECUTE_TEST_COVERALLS == 'true' ]]; then composer require --no-update satooshi/php-coveralls:dev-master ; fi
 
 install:
   - travis_retry composer install --no-interaction --prefer-source
```
Comment (UFOMelkor) [inline .travis.yml]: Does this have any effect after `composer update` in `before_install`?

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 3]
```diff
@@ -35,6 +36,7 @@
         'query' => QueryBus::class,
     ];
 
+
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: Two empty lines intended?

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 4]
```diff
@@ -118,51 +120,65 @@ private function busLoad(
      */
     private function loadBus(string $type, string $name, array $options, ContainerBuilder $container)
     {
+        $serviceBusName = sprintf('prooph_service_bus.%s', $name);
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: String concatenation is used 8 times while `sprintf` is used three times. Perhaps we can unify this to string concatenation?

Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: `$serviceBusId` would be more consistent.

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 5]
```diff
@@ -118,51 +120,65 @@ private function busLoad(
...
         $serviceBusDefinition = $container->setDefinition(
-            sprintf('prooph_service_bus.%s', $name),
+            $serviceBusName,
             new DefinitionDecorator('prooph_service_bus.' . $type . '_bus')
         );
 
+
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: Two empty lines intended?

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 6]
```diff
@@ -118,51 +120,65 @@ private function busLoad(
...
         if (!empty($options['plugins'])) {
             foreach ($options['plugins'] as $index => $util) {
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: Unused variable `$index`

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 7]
```diff
@@ -118,51 +120,65 @@ private function busLoad(
...
-            ->setArguments([new Reference($messageFactoryId)]);
+                $messageFactoryPluginDefinition
+            );
 
-        $serviceBusDefinition->addMethodCall('utilize', [new Reference($messageFactoryPluginId)]);
 
         // define router
+        $routerId = null;
         if (!empty($options['router'])) {
             $routerId = sprintf('prooph_service_bus.%s.router', $name);
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: See comment above about `sprintf`.

File: src/DependencyInjection/ProophServiceBusExtension.php [hunk 8]
```diff
@@ -118,51 +120,65 @@ private function busLoad(
...
+            $container->setDefinition($routerId, $routerDefinition);
         }
 
-        //Add container plugin
-        $serviceBusDefinition->addMethodCall('utilize', [new Reference('prooph_service_bus.container_plugin')]);
+        //Attach container plugin
+        $containerPluginId = 'prooph_service_bus.container_plugin';
+        $pluginIds = array_filter(array_merge($options['plugins'], [$containerPluginId, $messageFactoryPluginId, $routerId]));
+
+
```
Comment (UFOMelkor) [inline src/DependencyInjection/ProophServiceBusExtension.php]: Two empty lines intended?

File: src/Resources/config/command_bus.xml [hunk 9]
```diff
@@ -10,7 +10,10 @@
     </parameters>
 
     <services>
-        <service id="prooph_service_bus.command_bus" class="%prooph_service_bus.command_bus.class%" abstract="true" />
+        <service id="prooph_service_bus.command_bus" class="%prooph_service_bus.command_bus.class%" abstract="true" >
+            <factory service="prooph_service_bus.command_bus_factory" method="create"/>
+        </service>
+        <service id="prooph_service_bus.command_bus_factory" class="%prooph_service_bus.message_bus_factory.class%"/>
```
Comment (UFOMelkor) [inline src/Resources/config/command_bus.xml]: Not sure but factory might be set to `public=false`

File: src/Resources/config/event_bus.xml [hunk 10]
```diff
@@ -10,7 +10,10 @@
     </parameters>
 
     <services>
-        <service id="prooph_service_bus.event_bus" class="%prooph_service_bus.event_bus.class%" abstract="true" />
+        <service id="prooph_service_bus.event_bus" class="%prooph_service_bus.event_bus.class%" abstract="true" >
+            <factory service="prooph_service_bus.event_bus_factory" method="create"/>
+        </service>
+        <service id="prooph_service_bus.event_bus_factory" class="%prooph_service_bus.message_bus_factory.class%"/>
```
Comment (UFOMelkor) [inline src/Resources/config/event_bus.xml]: See comment above.

File: src/Resources/config/query_bus.xml [hunk 11]
```diff
@@ -10,7 +10,10 @@
     </parameters>
 
     <services>
-        <service id="prooph_service_bus.query_bus" class="%prooph_service_bus.query_bus.class%" abstract="true" />
+        <service id="prooph_service_bus.query_bus" class="%prooph_service_bus.query_bus.class%" abstract="true" >
+            <factory service="prooph_service_bus.query_bus_factory" method="create"/>
+        </service>
+        <service id="prooph_service_bus.query_bus_factory" class="%prooph_service_bus.message_bus_factory.class%"/>
```
Comment (UFOMelkor) [inline src/Resources/config/query_bus.xml]: See comment above.

File: composer.json [hunk 12]
```diff
@@ -71,7 +73,7 @@
   "extra": {
     "branch-alias": {
       "dev-master": "1.0-dev",
-      "dev-develop": "1.1-dev"
+      "dev-prooph_v7": "0.3-dev"
```
Comment (UFOMelkor) [inline composer.json]: As [v0.3 has been released](https://github.com/prooph/service-bus-symfony-bundle/releases/tag/v0.3) this should be `0.4-dev` (or completely removed after finishing?)

**Review by UFOMelkor — COMMENTED** [review]
Some small notes …

File: composer.json (hunk 12 above)
Comment (mablae) [reply composer.json]: I was a bit in struggle with these dependecies setup. Maybe the alias should be removed, true.

**Comment by mablae** [comment]
@UFOMelkor  Thank you for the review, I will adjust the stuff.

File: composer.json [hunk 13]
```diff
@@ -50,28 +50,29 @@
...
+    "prooph/service-bus": "~6.0",
     "container-interop/container-interop": "^1.1"
   },
   "require-dev": {
     "phpunit/phpunit": "^5.3",
     "phpunit/php-invoker": "^1.1.4",
     "fabpot/php-cs-fixer": "^1.11",
     "bookdown/bookdown": "1.x-dev as 1.0.0",
-    "tobiju/bookdown-bootswatch-templates": "1.0.x-dev"
+    "tobiju/bookdown-bootswatch-templates": "^1.0",
```
Comment (prolic) [inline composer.json]: prooph now has custom templates!

Comment (prolic) [reply composer.json]: see: https://github.com/prooph/bookdown-template

Comment (mablae) [reply composer.json]: Thanks, will apply that.

**Comment by mablae** [comment]
Ping @UFOMelkor , @codeliner , @prolic - Sorry I totally forgot about this branch. 

I just did some updates. Travis passes now and the wrong branch alias also was removed. 
Let me know if anything else is missing.

**Review by codeliner — APPROVED** [review]

**Comment by codeliner** [comment]
thx @mablae

**Pull request merged**